server/player/src/packet_handlers/inventory.rs

```diff
@@ -574,6 +574,36 @@ fn handle_number_key(
     Ok(())
 }
 
+fn handle_middle_click(
+    _game: &mut Game,
+    world: &mut World,
+    player: Entity,
+    packet: ClickWindow
+) -> anyhow::Result<()> {
+    let gamemode = *world.get::<Gamemode>(player);
+    if Gamemode::Creative == gamemode {
+        if let Some(_) = world.try_get::<PickedItem>(player) {
+            // Player already has something in its hand.
+            return Ok(());
+        }
+        let window = world.get::<Window>(player);
+        let accessor = window.accessor(world)?;
+
+        // Pick the item in the slot
+        let picked = accessor.item_at(packet.slot as usize)?;
+        let count = picked.map(|item| item.ty.stack_size()).unwrap_or(0);
```

Comment:
We cannot have a stack size of 0, should at least be one. Therefore if stack_size fails we should do nothing.

Reply:
In other words, just use unwrap. Since `stack_size()` can't return anything less than 1. You are welcome to update the return type of `Item::StackSize` to `NoneZeroU32`.

Reply:
Alright, nvm, I'm wrong. Due to the `picked` being `None` this should be ok.